airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3ConsumerFactory.java

```diff
@@ -94,19 +94,19 @@ private OnStartFunction onStartFunction(final BlobStorageOperations storageOpera
         if (writeConfig.getSyncMode().equals(DestinationSyncMode.OVERWRITE)) {
           final String namespace = writeConfig.getNamespace();
           final String stream = writeConfig.getStreamName();
-          final String outputBucketPath = writeConfig.getOutputBucketPath();
+          final String streamOutputPath = writeConfig.getFullOutputPath();
```

Comment:
The problem with this approach is:
1. First sync runs let's say on 2022-04-04 so the output path may be: `bucketpath/namespace/stream_name/2022/04/04/2022_04_04_123456_0.csv.gz`
2. Second sync runs let's say on 2022-04-05 so the output path may be: `bucketpath/namespace/stream_name/2022/04/05/2022_04_05_987654_0.csv.gz` and since we are in overwrite, we need to delete the previous folder/files before writing the new one

I already fixed the bug by running a recursive list on `bucketpath/` then regex match the path of files underneath with a pattern that looks like `bucketpath/namespace/stream_name/[0-9]{4}/[0-9]{2}/[0-9]{2}/[0-9]{2}/[0-9]{4}_[0-9]{2}_[0-9]{2}_[0-9]+_[0-9]+.csv.gz`

(I'm repeating the date twice for the sake of the example because it could be either part of the folder or part of the filename)